TrackingTools/TrackFitters/plugins/KFFittingSmootherESProducer.cc

```diff
@@ -199,11 +216,22 @@ namespace {
 #endif
 
       bool hasNaN = false;
-      if (!smoothed.isValid() || (hasNaN = !checkForNans(smoothed)) || (smoothed.foundHits() < theMinNumberOfHits)) {
+      double absTrajEta = 99.0;
+      if (smoothed.isValid()) {
+        absTrajEta = fabs(smoothed.lastMeasurement()
+                              .updatedState()
+                              .freeTrajectoryState()
+                              ->momentum()
+                              .eta());  //needed for eta-extended electrons
+      }
+      int thisHitCut = theMinNumberOfHits;
+      if (absTrajEta > theHighEtaSwitch)
+        thisHitCut = theMinNumberOfHitsHighEta;
+      if (!smoothed.isValid() || (hasNaN = !checkForNans(smoothed)) || (smoothed.foundHits() < thisHitCut)) {
```

Comment:
seems like code duplication with respect to L102-112. can you make this a function?